### DISCUSSION TITLE
looking for a precompiled bin - file for the for the lilygo s3 T-Display

### DISCUSSION THREAD

**Discussion by JohnnyTank** [micropython/micropython#10208] [ESP32] [answered]
Has anyone a working compiled version for the lilygo s3 t-Display.
I have no experience with C-Compilers and no linux-machine on hand.
I am using a windows 10 machine.
I have tried to flash various bin-Files with Thonny - sucessfull.
But when restarting  it hangs everytime at the usb-initialisation.

**Comment by mattytrentini** [comment]
#8689 has support for the ESP32-S3 variants with octal PSRAM. This includes the ESP32-S3R8 used in the [LilyGo T-Display S3](https://www.lilygo.cc/products/t-display-s3) - and there's even a pre-built firmware in the comments :)

**Reply by JohnnyTank** [reply]
Thank you for your answer!
I have tried this one.:
lilygo-t-embed-mpy-with7789.bin
Seems to work a little better - the board is now talking with Thonny but -
Thonny gives the error message:
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
invalid header: 0xffffffff
PROBLEM IN THONNY'S BACK-END: Internal error (serial.serialutil.SerialTimeoutException: Write timeout).
See Thonny's backend.log for more info.
You may need to press "Stop/Restart" or hard-reset your MicroPython device and try again.


Process ended with exit code 0.

**Comment by russhughes** [comment, marked as answer]
Take a look at https://github.com/russhughes/st7789s3_esp_lcd, https://github.com/russhughes/st7789s3_mpy, or https://github.com/russhughes/t-display-s3. 

The first one is the fastest, but it is new and not as tested as the second.  The third is written in python and has fewer features.

**Reply by JohnnyTank** [reply]
I have tried the firmware and it is doing the same as:
https://github.com/orgs/micropython/discussions/10208#discussioncomment-4393599

**Reply by russhughes** [reply]
You'll have to use another program to flash micropython. I just tried Thonny 4.0.1 and it tried to flash the esp32-s3 starting at 0x1000 instead of 0x0. [Expressif's Flash Download Tool for Windows](https://www.espressif.com/en/support/download/other-tools) worked for me.

**Reply by bixb922** [reply]
Power on with boot button pressed.

Then I used:

python -m esptool --chip esp32s3 --port com4 write_flash -z 0 firmware.bin

**Reply by JohnnyTank** [reply]
It was the Adress thank you very much for your help!
